usr/src/lib/brand/bhyve/zone/boot.c

```diff
@@ -271,8 +277,15 @@ add_disk(char *disk, char *path, char *slotconf, size_t slotconf_len)
 		 * be the most familiar name for the operation (and less
 		 * likely to cause confusion).
 		 */
-		if (is_env_string("device", disk, "notrim", "true"))
-			nodelstr = ",nodelete";
+		nodelete = is_env_true("device", disk, "notrim");
+		guest_block_size = get_zcfg_var("device", disk,
+		    "guest_block_size");
+
+		/* Treat a 0 size to mean the whatever the volume advertises */
+		if (guest_block_size != NULL &&
+		    strcmp(guest_block_size, "0") == 0) {
+			guest_block_size = NULL;
+		}
```

Comment:
This gives the same behavior as if `guest_block_size` is not specified, right?  The smartos-live change does not allow `0` as a value.  Why would we need it here?

Reply:
Correct. In some of the testing, despite the comments in proptable.js, I was still seeing instances where the property wasn't going away after setting it to zero. Since I wasn't sure how much I wanted to wade through all of that, I figured adding another safeguard couldn't hurt.